pkg/deploymanager/config.go

```diff
@@ -0,0 +1,120 @@
+package deploymanager
+
+import (
+	"fmt"
+	"os"
+
+	olmclient "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	lvmv1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// InstallNamespace is the namespace lvm is installed into
+const InstallNamespace = "openshift-storage"
+
+// DeployManager is a util tool used by the tests
+type DeployManager struct {
+	olmClient      *olmclient.Clientset
+	k8sClient      *kubernetes.Clientset
+	crClient       crclient.Client
+	parameterCodec runtime.ParameterCodec
+	lvmClient      *rest.RESTClient
+}
+
+// GetCrClient is the function used to retrieve the controller-runtime client.
+func (t *DeployManager) GetCrClient() crclient.Client {
+	return t.crClient
+}
+
+// GetK8sClient is the function used to retrieve the kubernetes client.
+func (t *DeployManager) GetK8sClient() *kubernetes.Clientset {
+	return t.k8sClient
+}
+
+// GetParameterCodec is the function used to retrieve the parameterCodec.
+func (t *DeployManager) GetParameterCodec() runtime.ParameterCodec {
+	return t.parameterCodec
+}
+
+// GetLvmClient is the function used to retrieve the lvm client.
+func (t *DeployManager) GetLvmClient() *rest.RESTClient {
+	return t.lvmClient
+}
+
+// NewDeployManager creates a DeployManager struct with default configuration.
+func NewDeployManager() (*DeployManager, error) {
+	codecs := serializer.NewCodecFactory(scheme.Scheme)
+	parameterCodec := runtime.NewParameterCodec(scheme.Scheme)
+
+	kubeconfig := os.Getenv("KUBECONFIG")
```

Comment:
so this var is being set in CI environment as well?

Reply:
You can  do like below (just example) if the kubeconfig is not set you can create client using the InClusterConfig

```go
import (
	"k8s.io/client-go/rest"
	"k8s.io/client-go/tools/clientcmd"
)

func NewK8sClient() (*kubernetes.Clientset, error) {
	kubeconfig := os.Getenv("KUBECONFIG")
	if kubeconfig != "" {
		cfg, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
		if err != nil {
			return nil, fmt.Errorf("failed to get cluster config from %q: %w", kubeconfig, err)
		}
	} else {
		cfg, err = rest.InClusterConfig()
		if err != nil {
			return nil, fmt.Errorf("failed to get cluster config: %w", err)
		}
	}
}

```

Reply:
yes, to run the tests.

Reply:
Madhu, can I ask where did you get this code from.

Reply:
https://github.com/kubernetes-csi/external-resizer/blob/d8b3fbccb596bb9a87e031d23face759e081fd54/cmd/csi-resizer/main.go#L107

Reply:
https://github.com/kubernetes-csi/external-provisioner/blob/daab12bcadd3489de2ae87d4395b279f1bb8f9e2/cmd/csi-provisioner/csi-provisioner.go#L155